src/Fakta/Agent.fs

```diff
@@ -1,78 +1,244 @@
 ﻿/// Agent can be used to query the Agent endpoints
 module Fakta.Agent
+open System
+open System.Text
+open Fakta
+open Fakta.Logging
+open Fakta.Impl
+open System
+open NodaTime
+open HttpFs.Client
+open Chiron
+
+let faktaAgentString = "Fakta.agent"
+
+let agentDottedPath (funcName: string) =
+  (sprintf "%s.%s" faktaAgentString funcName)
 
 /// CheckDeregister is used to deregister a check with the local agent
 let checkDeregister (state : FaktaState) (checkId : string) : Async<Choice<unit, Error>> =
-  raise (TBD "TODO")
+    let urlPath = (sprintf "check/deregister/%s" checkId)
+    let uriBuilder = UriBuilder.ofAgent state.config urlPath
+    let result = Async.RunSynchronously (call state (agentDottedPath "checkDeregister") id uriBuilder HttpMethod.Put)
+    async {
+      match result with 
+      | Choice1Of2 x -> 
+         return Choice1Of2 ()
+      | Choice2Of2 err -> return Choice2Of2(err)
+    }
 
 /// CheckRegister is used to register a new check with the local agent 
-let checkRegister (state : FaktaState) (checkId : string) : Async<Choice<unit, Error>> =
-  raise (TBD "TODO")
+let checkRegister (state : FaktaState) (checkRegistration : AgentCheckRegistration) : Async<Choice<unit, Error>> =
+    let urlPath = (sprintf "check/register")
+    let uriBuilder = UriBuilder.ofAgent state.config urlPath
+    let serializedCheckReg = Json.serialize checkRegistration |> Json.format
+    let result = Async.RunSynchronously (call state (agentDottedPath "checkRegister") (withJsonBody serializedCheckReg) uriBuilder HttpMethod.Put)
+    async {
+      match result with 
+      | Choice1Of2 x -> 
+         return Choice1Of2 ()
+      | Choice2Of2 err -> return Choice2Of2(err)
+    }
+
 
 /// Checks returns the locally registered checks
 let checks (state : FaktaState) : Async<Choice<Map<string, AgentCheck>, Error>> =
-  raise (TBD "TODO")
+    let urlPath = (sprintf "checks")
+    let uriBuilder = UriBuilder.ofAgent state.config urlPath
+    let result = Async.RunSynchronously (call state (agentDottedPath urlPath) id uriBuilder HttpMethod.Get)
+    async {
+      match result with 
+      | Choice1Of2 x -> 
+         let body, (dur:Duration, resp:Response) = x
+         let  item = if body = "" then Map.empty else Json.deserialize (Json.parse body)
+         return Choice1Of2 (item)
+      | Choice2Of2 err -> return Choice2Of2(err)
+    }
+
+  
+let setNodeMaintenanceMode (state : FaktaState) (enable : bool) : Async<Choice<unit, Error>> =
+    let urlPath = (sprintf "maintenance")
+    let uriBuilder = UriBuilder.ofAgent state.config urlPath 
+                     |> flip UriBuilder.mappendRange [ yield "enable", Some((enable.ToString().ToLower())) ]
```

Comment:
You don't need to allocate "True" and then "true" just to print the constant "true". It's better to do `if enable then yield "enable", Some "true"` than this, since it would save you two string allocations.